Created a Home Page with the Poll component

WesleyErasmus opened the pull request.

WesleyErasmus committed:
Created a Home Page with the Poll component

hirebarend commented on src/components/Poll.tsx:
```diff
@@ -0,0 +1,89 @@
+import "../styles/poll.css";
+
+// Ratio button vote options array
+const voteOptions = [
```
Instead of a constant value here, look into props of React components and add to the Poll

WesleyErasmus replied:
Would it be a good idea to have a reusable RadioInput.tsx component that I can import to my Poll.tsx component? 
<RadioInput value={"JavaScript"} label={"JavaScript"} />
<RadioInput value={"Python"} label={"Python"} />
<RadioInput value={"Go"} label={"Go"} />

I have been exploring that option but wondered if it would conflict with the idea of using props in the HomePage.tsx Poll component.

hirebarend replied:
You could, not always necessary to break components down to that level

hirebarend commented on src/components/Poll.tsx:
```diff
@@ -0,0 +1,89 @@
+const Poll = () => {
+
+  // Initialize a state variable to track the selected programming language. Set to an empty string
+  const [selectedLanguage, setSelectedLanguage] = useState("");
```
In software engineering, we try to solve the business problem rather than a specific example. This poll component might be responsible for a "Programming Language" poll now, but in future it could be for your favorite pizza place

hirebarend commented on src/components/Poll.tsx:
```diff
@@ -0,0 +1,89 @@
+      <form className="poll-form" action="">
+        {/* Form heading */}
+        <div className="form-heading">
+          What programming languages do you use during interviews?
```
Possibly move this to props as well

hirebarend commented on src/styles/poll.css:
```diff
@@ -0,0 +1,109 @@
+/* Styling applies to all 'label' semantic HTML tags */
+label {
```
More of a side note, you are welcome to continue with your own styling, but you could try a framework like Bootstrap or Tailwindcss. Let me know if you want help to setup either one of them

WesleyErasmus committed:
Poll.tsx props and naming updates

Switched to using props for the component heading and the radio inputs.

Updated state naming to be reusable

WesleyErasmus committed:
Poll.tsx - removed key attributes from labels as no longer using array map() function

WesleyErasmus committed:
Poll.tsx - updated React state name

Refactored React state variable and function names to allow reusability across different poll questions
